src/lib/dialog/dialog-directives.ts

```diff
@@ -0,0 +1,49 @@
+import {Directive} from '@angular/core';
+import {MdDialog} from './dialog';
+
+
+/**
+ * Button that will close the current dialog.
+ */
+@Directive({
+  selector: 'button[md-dialog-close]',
+  host: {
+    '(click)': 'dialog.closeTop()'
+  }
+})
+export class MdDialogClose {
+  constructor(public dialog: MdDialog) { }
+}
+
+/**
+ * Title of a dialog element. Stays fixed to the top of the dialog when scrolling.
+ */
+@Directive({
+  selector: '[md-dialog-title]',
+  host: {
+    role: 'heading'
+  }
+})
+export class MdDialogTitle { }
+
+
+/**
+ * Scrollable content container of a dialog.
+ */
+@Directive({
+  selector: '[md-dialog-content], md-dialog-content',
+  host: {
+    role: 'main'
```

Comment:
I don't think we should have this role; nothing in [the W3C authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal) mentiones it and the docs for [role="main"](https://www.w3.org/TR/wai-aria/roles#main) says there should only be one per "document" (for which a modal dialog doesn't count).